pkg/provision/source.go

```diff
@@ -83,8 +90,15 @@ func (s *pollSource) Reservations(ctx context.Context) <-chan *Reservation {
 			case <-ctx.Done():
 				return
 			default:
-				for _, r := range res {
-					ch <- r
+				for idx, r := range res {
+					reservation := ReservationFromSource{
+						r,
+						false,
+					}
+					if idx == len(res)-1 {
```

Comment:
I don't think this condition is correct. 
Isn't it better to do it like `if lastID == reservation.ID` ?
You also need to make sure that you only set the `last` field to true only once and once only for the full life duration of provisiond. The other time it will be trigger by the cron

Reply:
LastID is an int and `reservation.ID` is a string. We cannot match on this.

Reply:
> You also need to make sure that you only set the `last` field to true only once and once only for the full life duration of provisiond. The other time it will be trigger by the cron

I don't understand what you mean by this.

Reply:
> Length of an array - 1 points to the last element in that array no? 
it does but nothing tell you the explorer send you all the workloads in one request. So unless you have process the reservation that has the lastID reported by the explorer, you have not reach the end.

>LastID is an int and reservation.ID is a string. We cannot match on this.

I think you can: it's generated like this https://github.com/threefoldtech/zos/blob/7727b34f2d71c8ac0cee5bb97de11461bceb136a/pkg/provision/primitives/converter.go#L252